packages/ckeditor5-find-and-replace/src/findandreplaceutils.ts

```diff
@@ -52,6 +52,21 @@ export default class FindAndReplaceUtils extends Plugin {
 	): Collection<ResultType> {
 		const results = startResults || new Collection();
 
+		const checkIfResultAlreadyOnList = ( marker: Marker ) => results.find(
```

Comment:
> I guess there may be a situation where we removed something that already had a search marker and we re-add both the search marker and the new search result?
 
Yes, there can be such case. It happens quite often in T&C mode when you remove multiple paragraphs at once and then undo it. You are absolutely right about performance, it will affect it, but at this moment it's still better than having the feature broken at all. I suspect that this function and huge part of implementation of `_onDocumentChange` will be changed anyway, and it'll be no longer needed after fixing changes detection.